ultimatepython/data_structures/linkedlist.py

```diff
@@ -0,0 +1,117 @@
+class Node:
+    def __init__(self, dataval):
+        self.dataval = dataval
+        self.nextval = None
+        self.preval = None
+
+class CLinkedList:
+    def __init__(self):
+        self.headval = None
+
+    def AtBeginning(self, newdata):
```

Comment:
I like what is being implied with `AtBeginning`. The problem is that the method does not have a verb in it which means it misses the intent. Furthermore, Python methods are usually written as `this_kind_of_style`.

I'd like to propose the following option: `insert_left` which seems to be what you're going for.